src/App.tsx

```diff
@@ -14,11 +15,74 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: [...goodsFromServer],
+    isShowed: false,
+  };
+
+  showGoods = () => {
+    this.setState({
+      isShowed: true,
+    });
+  };
+
+  revers = () => {
+    this.setState({
+      goods: [...goodsFromServer].reverse(),
+    });
+  };
+
+  sortAlplabetically = () => {
+    this.setState({
+      goods: [...goodsFromServer].sort((a, b) => a.localeCompare(b)),
+    });
+  };
+
+  reset = () => {
+    this.setState({
+      goods: [...goodsFromServer],
```

Comment:
Again, can be used without copy.
```suggestion
      goods: goodsFromServer,
```

Reply:
Still not fixed.